edx_analytics_transformers/django/models.py

```diff
@@ -0,0 +1,165 @@
+"""
+Models for filtering of events
+"""
+import logging
+
+from django.db import models
+from model_utils.models import TimeStampedModel
+from jsonfield.fields import JSONField
+
+from edx_django_utils.cache import TieredCache, get_cache_key
+
+
+logger = logging.getLogger(__name__)
+
+
+ROUTER_CACHE_NAMESPACE = 'edx_analytics.request_router'
+
+
+def get_value_from_dotted_path(dict_obj, dotted_key):
+    """
+    Map the dotted key to nested keys for dict and return the matching value.
+
+    For example:
+        'key_a.key_b.key_c' will look for the folowing value:
+
+        {
+            'key_a': {
+                'key_b': {
+                    'key_c': 'final value'
+                }
+            }
+        }
+
+    Arguments:
+        dict_obj (dict)  :    dictionary for which the value is required
+        dotted_key (str) :    dotted key string for the dict
+
+    Returns:
+        ANY :                 depends upon the value in dict
+    """
+    nested_keys = dotted_key.split('.')
+    result = dict_obj
+    try:
+        for key in nested_keys:
+            result = result[key]
+    except KeyError:
+        return None
+    return result
+
+
+class RouterConfiguration(TimeStampedModel):
+    """
+    Configurations for filtering and then routing events to hosts.
+    """
+
+    backend_name = models.CharField(
+        max_length=50,
+        verbose_name='Backend name',
+        null=False,
+        blank=False,
+        help_text=(
+            'Name of the tracking backend on which this router should be applied.'
+            '<br/>'
+            'Please note that this field is <b>case sensitive.</b>'
+        )
+    )
+
+    is_enabled = models.BooleanField(
+        default=True,
+        verbose_name='Is Enabled'
+    )
+
+    configurations = JSONField()
```

Comment:
This can contain credentials and so should be encrypted at rest using something like https://django-fernet-fields.readthedocs.io/en/latest/.

Reply:
Updated